core/src/main/java/org/apache/gravitino/GravitinoEnv.java

```diff
@@ -537,14 +539,28 @@ private void initGravitinoServerComponents() {
 
     SchemaOperationDispatcher schemaOperationDispatcher =
         new SchemaOperationDispatcher(catalogManager, entityStore, idGenerator);
-    SchemaHookDispatcher schemaHookDispatcher = new SchemaHookDispatcher(schemaOperationDispatcher);
+    SchemaDispatcher schemaDispatcherProxy =
+        (SchemaDispatcher)
+            Proxy.newProxyInstance(
+                SchemaDispatcher.class.getClassLoader(),
+                new Class[] {SchemaDispatcher.class},
+                new OperationDispatcherInterceptor(
```

Comment:
I'm not so familar with this interceptor mechanism, how does it reduce the number of calling?

Reply:
My question is why using reflection/proxy can reduce the number of calls.

Reply:
> When will this public Object invoke(Object proxy, Method method, Object[] args) throws Throwable be invoked?

When any interface method of `OperationDispatcher` is called, the interceptor method will be triggered, like `loadTable`.